app/controllers/movies_controller.rb

```diff
@@ -0,0 +1,22 @@
+class MoviesController < ApplicationController
+
+  def show
+    @movie = Movie.find(params[:id])
+    @actors = @movie.actors.order(age: :asc)
+    @average_age = @actors.average(:age)
+  end
+
+  def add_actor
+    @movie = Movie.find(params[:id])
+    actor = Actor.find_by(id: params[:movie][:actor_id]) # Find the actor by ID, returns nil if not found
+
+    if actor
+      @movie.actors << actor
+    else
+      flash[:alert] = "Actor with ID #{params[:movie][:actor_id]} not found."
+    end
+    
+    redirect_to "/movies/#{@movie.id}"
+  end
```

Comment:
This `add_actor` method should probably be called `create` and this `create` method could be placed in 1 of 2 places; either a `ActorMoviesController` or the `MoviesController` like you have it here but you would need to check to see if the `params` included an actor ID before making the join. 

I think the easiest and most straightforward way to do this would just be in the `ActorMoviesController`. While this isn't an MVC infraction, I think it could have been organized better.